### DISCUSSION TITLE
Can't create pull request

### DISCUSSION THREAD

**Discussion by juhanjuku** [OHF-Voice/intents#1432] [_Q&A]
I'm trying second time to add new language 'et' 'Eesti'. I use CodeSpace and follow tutorial at
https://github.com/home-assistant/intents/blob/main/docs/codespace/README.md
I made minimal changes, as suggested in tutorial, all tests passed 100%
Last step > Source control > "Commit & Create Pull Request" , 
but I don't have such choice?

![codespace](https://github.com/home-assistant/intents/assets/12792066/dc179935-4960-48b2-a90d-22fd1be2ba53)

My first attempt went good, I managed to create pull request #1375
It waited acceptance so long that meantime merge conflict appeared in languages.yaml (language 'eu' was added)
I tried to resolve this conflict but gave up, my GitHub skills were not good enough.
So I closed my first pull request, deleted 'intents' from my GitHub account and started from zero.

But now I don't have choice to create pull request and I can't understand what I'm doing wrong.
As last hope, I created new GitHub account and repeated all steps there > same result > no choice to create pull request.

I really like to add my contribution, but I'm stuck.

**Comment by tetele** [comment]
> My first attempt went good, I managed to create pull request https://github.com/home-assistant/intents/pull/1375

You didn't create a feature branch in your fork, though. That's OK.

> As last hope, I created new GitHub account and repeated all steps there > same result > no choice to create pull request.

That's excessive. The ability to create PRs in VSCode is not linked to your account. There may be some issue with the repo, your fork or local copy of the repo.

I don't know why you can't create a PR when committing, but try this:
- create a branch from `main` in your repo (call it `es-language` for example)
- create a commit with your code (without pushing or creating a PR yet)
- publish/sync your branch (first time you do this, it will create a new branch called `es-language` on your repo, `juhanjuku/intents` or whatever it's called. The next times, it will simply push the commits from your codespace to the already existing branch of your repo)
- go to the Github tab in the sidebar and click "Create pull request" (see screenshot below)
- follow the rest of the instructions from the tutorial you linked

![github PR](https://github.com/home-assistant/intents/assets/920098/935a28f4-f5c6-48f8-90d5-c9806204a335)

If this doesn't work, please get back with further info. Your initial post is one of the best examples of "I tried following this, I tried this and here's where I got stuck", so keep that up!

**Comment by juhanjuku** [comment]
> > My first attempt went good, I managed to create pull request #1375
> 
> You didn't create a feature branch in your fork, though. That's OK.
> 
Sorry, but I cant' see such step in tutorial

> > As last hope, I created new GitHub account and repeated all steps there > same result > no choice to create pull request.
> 
> That's excessive. The ability to create PRs in VSCode is not linked to your account. There may be some issue with the repo, your fork or local copy of the repo.
> 
> I don't know why you can't create a PR when committing, but try this:
> 
> * create a branch from `main` in your repo (call it `es-language` for example)
> * create a commit with your code (without pushing or creating a PR yet)
> * publish/sync your branch (first time you do this, it will create a new branch called `es-language` on your repo, `juhanjuku/intents` or whatever it's called. The next times, it will simply push the commits from your codespace to the already existing branch of your repo)
> * go to the Github tab in the sidebar and click "Create pull request" (see screenshot below)
> * follow the rest of the instructions from the tutorial you linked
> 
> ![github PR](https://user-images.githubusercontent.com/920098/247484310-935a28f4-f5c6-48f8-90d5-c9806204a335.png)
> 
> If this doesn't work, please get back with further info. Your initial post is one of the best examples of "I tried following this, I tried this and here's where I got stuck", so keep that up!

Tried to create CodeSpace on main several times, but unfortunately I don't have all buttons as on your picture - GitHub button is missing?

![2023-06-21_213554](https://github.com/home-assistant/intents/assets/12792066/b12e6e86-0fcd-46e3-a1c2-5c6b59b0aecc)

Sorry, to be such noob with GitHub. 
Tomorrow my son comes to visit us and I hope he can help me to get on track :)

**Comment by tetele** [comment]
I think you no longer have a repo tied to your codespace...

Go to https://github.com/codespaces and delete your codespace
Fork the intents repo
On your fork, create a new codespace
In your codespace, create a branch
If all went well so far, follow my instructions above or the tutorial
Report back

**Comment by juhanjuku** [comment]
Now I managed to create Pull request #1435

**Comment by steveyegge** [comment]
Thanks for reporting this! We investigated and found this was indeed a real problem - even in our own projects we had significant rates of empty descriptions:

- beads project: 110/630 issues (17.5%)
- wyvern project: 8/119 issues (6.7%)
- vc project: 3/170 issues (1.8%)

We've shipped a two-part solution in commit bf74160:

**1. Documentation improvements** - Updated AGENTS.md, MCP tool docstrings, and Copilot instructions with:
- Prominent guidance section explaining why descriptions matter
- Good vs bad examples showing what to include (why/what/how)
- All code examples now include `--description` flags

**2. Friendly validation warnings** - When creating issues without descriptions:
```
⚠ Creating issue without description.
  Issues without descriptions lack context for future work.
  Consider adding --description="Why this issue exists and what needs to be done"
```
- Works in both CLI and daemon modes
- Smart enough to skip test issues
- Non-blocking (users can still proceed)

The warning approach is similar to how we handle test issues - gentle reminders rather than strict enforcement.

For the hooks/skills approach you mentioned, we've created bd-b0c8 to research that as a potential enhancement, but the documentation + warning combo should help significantly in the meantime.

Thanks again for the detailed report - this will help everyone using beads with Claude Code!
